rebuttal.tex

```diff
@@ -0,0 +1,730 @@
+\begin{solution}
+This suggestion was implemented.
+\end{solution}
+
+\question Line 355, add a sentence explaining that in SymPy \texttt{str == repr}, because in Python \texttt{repr} is used to get an unambiguous valid Python code representation, while the return value of \texttt{str} is meant to be human-readable.
+\begin{solution}
+
+\end{solution}
+
+\question Lines 359 and 379: what is 2D text representation? It seems that ``2D'' shouldn't be here.
+\begin{solution}
+
+\end{solution}
+
+\question Line 427: every dictionary is a ``dictionary of keys''. This should be a dictionary with coordinate tuples as keys associated with the appropriate values.
+\begin{solution}
+``Dictionary of Keys'' (DOK) refers specifically to a sparse matrix
+representation where entries are stored as \texttt{(row, column)} pairs
+mapping to the elements. See for instance \texttt{scipy.sparse.dok\_sparse}. We
+have fixed the capitalization of ``Dictionary of Keys'' and added the
+abbreviation (DOK), and stated explicitly what is meant by this. We have also
+made a similar note about List of Lists (LIL) earlier in the section.
+\end{solution}
+
+\question Section 4 would benefit from an introduction, and lines 440-- should become a new subsection 4.1. (named ``Float'' or ``Real numbers support'' or similar).
+\begin{solution}
+
+\end{solution}
+
+\question I suggest a better example for lines 459--460: a computation of $(e^{100} +1)− e^{100}$:
+
+\begin{verbatim}
+   >>> (exp(100)+1).evalf() - exp(100).evalf()
+   0
+   >>> ((exp(100)+1) - exp(100)).evalf()
+   1.00000000000000
+   >>> (exp(100)+1) - exp(100)
+   1
+\end{verbatim}
+
+or two different ways to compute the 100th Fibonacci number:
+
+\begin{verbatim}
+   >>> phi = (1+sqrt(5))/2
+   >>> psi = (1-sqrt(5))/2
+   >>> ((phi**100-psi**100)/sqrt(5)).evalf()-fibonacci(100)
+   65536.0000000000
+   >>> ((phi**100-psi**100)/sqrt(5) - fibonacci(100)).evalf()
+   0.e-104
+\end{verbatim}
+
+Obviously, like your own example, these are problematic because a part of the computation is relying on Python's builtin floats. Please include a comment on whether symbolic computation (i.e., applying \texttt{evalf()} on the whole expression) always avoids these errors or not, possibly with an example when it doesn't resolve this problem.
+\begin{solution}
+First your example, unfortunately, doesn't work due to automatic
+simplification (see last input line --- \texttt{exp(100)} cancel).
+Second example is good, but our looks to be more simple.
+
+Applying \texttt{evalf()} has nothing related to \emph{symbolic} computation,
+that method is for numerical evaluation, that should be clear from
+the sentence below ``The evalf method converts a constant symbolic expression
+to a Float\dots''.  Numerical evaluation of the whole expression should
+always solve cancellation problem, as it was stated in the sentence
+``Applying the evalf method to the whole expression solves this problem.''
+\end{solution}
```

Comment:
If so, can you suggest how to improve the wording in the article, and if not, do you have a counterexample?